rfcs/experimental/kernel_templates/README.md

```diff
@@ -0,0 +1,292 @@
+For example, a user program may manage a reusable memory pool.
+Also, this is especially useful when an algorithm is called in a loop and the allocation overhead
+cannot be amortized by memory pools provided by the device or compiler runtime.
+
+It may be preferable to make this the only option and
+avoid internal global memory allocations altogether.
+This approach eliminates the overhead associated with `host_task`,
+which retains temporary memory until an algorithm completes its asynchronous execution.
+Additionally, it reduces the number of interfaces that need to be supported.
+
+This should be done in conjunction with the
+[Global and Local Memory Requirements](#reporting-global-and-local-memory-requirements) interface.
+
+### Asynchronous Execution
+
+The algorithms return a `sycl::event` but do not accept any input events.
+As a result, they are not fully asynchronous.
+
+It should be investigated whether full asynchrony is desirable,
+and if so, how to implement it correctly.
+
+### Separation of Specializations Based on a Problem Size
+
+Different approaches may be more effective when processing different number of elements.
+For example, a small number of elements can be processed by a single work-group.
+
+Separating such cases would reduce compilation time.
+
+### Algorithms to Implement
+
+A list of algorithms to implement should be defined.
+
+### Benchmarking
+
+A benchmark suite can help select the best configuration and parameters
+for a given workload and hardware.
+
+### Default Kernel Configuration Values
+
+`KernelParam` template parameter
+is placed before all the deduced parameters to simplify
+a potential addition of a default value in the future.
+However, it is unclear if a default value should be provided,
+especially given the extension's focus on performance tuning.
+
+Creation of a default value with optimal settings is impossible right now.
+`kernel_param` template argument, which substitutes `KernelParam`,
+includes compile-time values, whose optimal selection depends on run-time information,
+such is the number of elements to process.
+
+[Runtime Parameters](kernel_configuration/README.md#runtime-parameters) raises a question
+of providing them at run-time.
+However, doing so would likely require querying device properties, introducing overhead.
+
+The complexity of manual selection of these values may be reduced by the [Benchmarks](#benchmarking),
+making default values less relevant.
+
+### Configuring an Algorithm with Multiple Kernels
+
+It needs to be explored how to configure algorithms with multiple performance-critical kernels.
+The `kernel_param` targets a single kernel, and currently, only one instance can be passed.
+
+### Compiler Extension and Differentiation of Algorithms
+
+Currently, it is unclear if we should defferentiate algorithms
+which rely compiler extensions.
+
+## Exit Criteria
+
+The proposed set of algorithms should transition to an extension if:
```

Comment:
Here, again, let's speak about transition into production, leaving aside whether it's "an extension" or something else.